xds/internal/server/rds_handler_test.go

```diff
@@ -0,0 +1,423 @@
+// +build go1.12
+
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package server
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/xds/internal/testutils/fakeclient"
+	"google.golang.org/grpc/xds/internal/xdsclient"
+)
+
+var (
+	route1 = "route1"
+	route2 = "route2"
+	route3 = "route3"
+)
+
+// setupTests creates a rds handler with a fake xds client for control over the
+// xds client.
+func setupTests(t *testing.T) (*rdsHandler, *fakeclient.Client) {
+	xdsC := fakeclient.NewClient()
+	rh := newRDSHandler(&listenerWrapper{xdsC: xdsC})
+	return rh, xdsC
+}
+
+// TestSuccessCaseOneRDSWatch tests the simplest scenario: the rds handler
+// receives a single route name, starts a watch for that route name, gets a
+// successful update, and then writes an update to the update channel for
+// listener to pick up.
+func (s) TestSuccessCaseOneRDSWatch(t *testing.T) {
+	rh, fakeClient := setupTests(t)
+	// When you first update the rds handler with a list of a single Route names
+	// that needs dynamic RDS Configuration, this Route name has not been seen
+	// before, so the RDS Handler should start a watch on that RouteName.
```

Comment:
Is this comment required? You *do*  have a function level comment which says the same in a more succinct way.